lib/api/candles.js

```diff
@@ -99,13 +102,10 @@ module.exports = function (app) {
     // Private
 
     var validExchange = function (e) {
-        if (e === 'poloniex') {
-            return poloniex;
-        } else if (e === 'bittrex') {
-            return bittrex;
-        } else {
-            return 'error';
-        }
+        if (!_.has(config.marketWatcher.exchanges, e) || !config.marketWatcher.exchanges[e] || !exchanges[e]) {
```

Comment:
Exchange must exists in config, be enabled in config and be in `exchanges` object (line 10).